src/lib/Ska/Starcheck/Obsid.pm

```diff
@@ -760,6 +760,8 @@ sub large_dither_checks {
     my $all_dither = shift;
     my $time_tol = 10;         # Commands must be within $time_tol of expectation
 
+    my $warn_free = 1;
```

Comment:
You need a new variable, but you can do it with fewer code lines by setting a variable to the length of `@{$self->{warn}}` and then later .. `if ($#{$self->{warn}} == $n_warn) {` (or whatever the right syntax).